### PR TITLE
[TF-TRT] Variable converters follow-up: INT8 support and edge cases

### PR DESCRIPTION
**Pull request by Nyrio**
This PR is a follow-up of #55337 and #55608. It includes tests and fixes for the following cases:

- INT8 support
- Variables shared by multiple engines
- `BiasAdd`, `BatchMatMul`, and binary ops with two constant inputs
- Segments with only resource inputs

### CODE REVIEW

**Comment by gbaned** [comment]
@Nyrio This PR is in draft, any update on this? Please. Thank you!

**Comment by Nyrio** [comment]
@bixia1  Following the merge of #55608, this one is now up for review.

File: tensorflow/python/compiler/tensorrt/test/testdata/gen_tf_variablev2_copy_model.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import numpy as np
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/testdata/gen_tf_variablev2_copy_model.py:1]: Please add copyright header.

File: tensorflow/python/compiler/tensorrt/test/testdata/gen_tf_variablev2_copy_model.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+from tensorflow.python.saved_model import signature_constants
+from tensorflow.python.saved_model import signature_def_utils
+from tensorflow.python.saved_model import tag_constants
+from tensorflow.python.saved_model import utils
+
+ops.disable_eager_execution()
+variable_scope.disable_resource_variables()
+
+def GenerateModelWithVariableV2_1Var2Engines(tf_saved_model_dir):
+  """Generate a model to test 1 VariableV2 node used in 2 engines."""
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/testdata/gen_tf_variablev2_copy_model.py:22]: s/Generate/Generates/

File: tensorflow/python/compiler/tensorrt/test/testdata/gen_tf_variablev2_copy_model.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+  def SimpleModel():
+    """Define model with a TF graph."""
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/testdata/gen_tf_variablev2_copy_model.py:25]: s/Define/Defines/

File: tensorflow/python/compiler/tensorrt/test/variables_test.py [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+from tensorflow.python.ops import math_ops
+from tensorflow.python.ops import nn
+from tensorflow.python.ops import variables
+from tensorflow.python.platform import tf_logging as logging
+from tensorflow.python.platform import test
+from tensorflow.python.saved_model import save, signature_constants
+
+
+def run_without_freezing():
+  """Execute test with model freezing disabled.
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/variables_test.py:38]: s/Execute/Executes/

File: tensorflow/python/compiler/tensorrt/test/variables_test.py [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+    @functools.wraps(f)
+    def decorated(self, *args, **kwargs):
+      with test_utils.experimental_feature_scope("disable_graph_freezing"):
+        f(self, *args, **kwargs)
+    return decorated
+  return decorator
+
+
+def run_all_without_freezing():
+  """Execute all tests in a class with model freezing disabled."""
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/variables_test.py:52]: s/Execute/Executes/

File: tensorflow/python/compiler/tensorrt/test/variables_test.py [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+  return test_util.for_all_test_methods(run_without_freezing)
+
+
+class TwoEnginesOneVarModel(module.Module):
+  """Model with two engines using a common variable.
+  The engines comprise ops that require constant inputs, so this checks that
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/variables_test.py:58]: maybe s/so this checks/the tests check/ ?

File: tensorflow/python/compiler/tensorrt/test/variables_test.py [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+  the variable is appropriately passed as weights to their converters in this
+  situation.
+  """
+  # TODO: figure out why the two engines seem to have resource inputs
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/variables_test.py:62]: Please add a period to the end of the TODO.

File: tensorflow/python/compiler/tensorrt/test/variables_test.py [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+
+  def GetParams(self):
+    shapes = [[2, 32, 32, 32, 3]]
+    my_model = TwoEnginesOneVarModel()
+    return self.BuildParams(my_model.__call__, dtypes.float32,
+                            input_shapes=shapes,
+                            output_shapes=shapes)
+
+  def ExpectedEnginesToBuild(self, run_params):
+    """Return the expected engines to build."""
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/variables_test.py:122]: s/Return/Returns/

File: tensorflow/python/compiler/tensorrt/test/variables_test.py [hunk 9]
```diff
@@ -0,0 +1,300 @@
...
+    return {
+        "TRTEngineOp_000": ["conv1", "bias1", "bias_add1", "relu1"],
+        "TRTEngineOp_001": ["conv0", "bias0", "bias_add0", "relu0"]
+    }
+
+
+class BiasAddTwoConstInputsModel(module.Module):
+  """Model with bias add taking only variable and constants inputs.
+  If the model were frozen, this would be constant-folded. This checks that
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/variables_test.py:131]: s/This/The tests/ ?

File: tensorflow/python/compiler/tensorrt/test/variables_test.py [hunk 10]
```diff
@@ -0,0 +1,300 @@
...
+  def ExpectedEnginesToBuild(self, run_params):
+    """Return the expected engines to build."""
+    return {
+        "TRTEngineOp_000": ["v1", "bias", "bias_add", "add", "relu"],
+    }
+
+
+class BatchMatMulTwoConstInputsModel(module.Module):
+  """Model with batched matmul taking only variable and constants inputs.
+  If the model were frozen, this would be constant-folded. This checks that
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/variables_test.py:186]: s/This/The tests/ ?

File: tensorflow/python/compiler/tensorrt/trt_convert_test.py [hunk 11]
```diff
@@ -1214,5 +1219,69 @@ def testReadVariableOp(self):
     self._TestVariableHelper("ReadVariableOp", "tf_readvariableop_saved_model",
                              "tftrt_readvariableop_saved_model", "output_0")
 
+  @test_util.run_v2_only
+  def testReadVariableOpInt8(self):
+    """Test conversion of ReadVariableOp nodes in INT8 precision."""
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/trt_convert_test.py:1224]: s/Test/Tests/

File: tensorflow/python/compiler/tensorrt/trt_convert_test.py [hunk 12]
```diff
@@ -1214,5 +1219,69 @@ def testReadVariableOp(self):
...
+
+    self._TestVariableHelper("ReadVariableOp",
+                             "tf_readvariableop_saved_model",
+                             "tftrt_readvariableop_saved_model",
+                             "output_0", "INT8")
+
+  def testVariableV2_1Var2Engines(self):
+    """Special."""
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/trt_convert_test.py:1232]: Can you expand this doc string?

File: tensorflow/python/compiler/tensorrt/trt_convert_test.py [hunk 13]
```diff
@@ -1253,7 +1253,7 @@ def testVariableV2_1Var2Engines(self):
 
     def input_fn():
       for shapes in input_shapes:
-        # return a list of input tensors
+        # Return a list of input tensors
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/trt_convert_test.py:1256]: Please add a period to the end.

**Review by bixia1 — APPROVED** [review]

**Comment by gbaned** [comment]
Hi @Nyrio Can you please resolve conflicts? Thank you!

File: tensorflow/compiler/tf2tensorrt/convert/convert_graph.cc [hunk 14]
```diff
@@ -196,7 +197,34 @@ Status GetEngineInfo(const Graph* g,
           // Already added before.
           continue;
         }
-        VLOG(1) << "Adding const node " << input_node->name();
+        VLOG(1) << "Adding const or variable node " << input_node->name();
+      } else if (input_node->type_string() == "Identity") {
+        // If one or a chain of Identity nodes originate from a constant or
+        // variable node, copy all to the graphdef.
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_graph.cc:203]: s/all/the Identify nodes/

File: tensorflow/compiler/tf2tensorrt/convert/convert_graph.cc [hunk 15]
```diff
@@ -447,10 +475,18 @@ Status CreateTRTNode(const TRTOptimizationPass::ConversionParams& params,
       }
     }
   }
+  int trt_inputs = 0;
+  for(auto& input: inputs) {
+    if(input.data_type != DT_RESOURCE) {
+      trt_inputs++;
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_graph.cc:481]: You may want to use absl::c_for_each to rewrite this.

File: tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc [hunk 16]
```diff
@@ -656,21 +656,26 @@ void TRTEngineOp::ExecuteCalibration(OpKernelContext* ctx,
   DummyAsyncHelper dummy_async_helper;
 
   CalibrationContext* calib_ctx = cache_res->calib_ctx_.get();
-  const int num_inputs = ctx->num_inputs();
+  int num_inputs = calib_ctx->device_tensors_.size();
+  int j = 0;
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc:660]: Can you replace j with a more meaning variable name?

File: tensorflow/compiler/tf2tensorrt/segment/segment.cc [hunk 17]
```diff
@@ -1152,7 +1152,28 @@ Status SegmentGraph(const Graph* tf_graph,
           que->pop_front();
           if (!visited.insert(node).second) continue;
           segment_nodes.erase(node);
-          for (auto in : (is_input_nodes || node->type_string() == "Const")
+
+          // We need to make sure that an identity that follows a variable is
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/segment/segment.cc:1156]: s/identity that/ identity node/?

File: tensorflow/python/compiler/tensorrt/test/variables_test.py [hunk 18]
```diff
@@ -0,0 +1,300 @@
...
+    saved_model_dir = self._GetSavedModelDir(
+        run_params, trt_test.GraphState.ORIGINAL)
+    logging.info("Saving input SavedModel to %s", saved_model_dir)
+    save.save(my_model, saved_model_dir,
+              {signature_constants.DEFAULT_SERVING_SIGNATURE_DEF_KEY: cfunc})
+    return saved_model_dir
+
+  def ShouldRunTest(self, run_params):
+    # Variables are only supported in dynamic shape mode.
+    # TODO: dynamic engine / static engine?
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/variables_test.py:110]: s/TODO/TODO(lsugy)/

File: tensorflow/python/compiler/tensorrt/test/variables_test.py (hunk 8 above)
Comment (bixia1) [reply tensorflow/python/compiler/tensorrt/test/variables_test.py:122]: Can you fix this?

File: tensorflow/python/compiler/tensorrt/test/variables_test.py [hunk 19]
```diff
@@ -0,0 +1,300 @@
...
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/variables_test.py:166]: s/TODO/TODO(lsugy)/

File: tensorflow/python/compiler/tensorrt/test/variables_test.py [hunk 20]
```diff
@@ -0,0 +1,300 @@
...
+
+  def GetParams(self):
+    shapes = [[8, 4, 4, 3]]
+    my_model = BiasAddTwoConstInputsModel()
+    return self.BuildParams(my_model.__call__, dtypes.float32,
+                            input_shapes=shapes,
+                            output_shapes=shapes)
+
+  def ExpectedEnginesToBuild(self, run_params):
+    """Return the expected engines to build."""
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/variables_test.py:178]: s/Return/Returns/

File: tensorflow/python/compiler/tensorrt/test/variables_test.py [hunk 21]
```diff
@@ -0,0 +1,300 @@
...
+    saved_model_dir = self._GetSavedModelDir(
+        run_params, trt_test.GraphState.ORIGINAL)
+    logging.info("Saving input SavedModel to %s", saved_model_dir)
+    save.save(my_model, saved_model_dir,
+              {signature_constants.DEFAULT_SERVING_SIGNATURE_DEF_KEY: cfunc})
+    return saved_model_dir
+
+  def ShouldRunTest(self, run_params):
+    # Variables are only supported in dynamic shape mode.
+    # TODO: dynamic engine / static engine?
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/variables_test.py:222]: s/TODO/TODO(lsugy)/

File: tensorflow/python/compiler/tensorrt/test/variables_test.py [hunk 22]
```diff
@@ -0,0 +1,300 @@
...
+
+  def GetParams(self):
+    shapes = [[8, 4, 4, 3]]
+    my_model = BatchMatMulTwoConstInputsModel()
+    return self.BuildParams(my_model.__call__, dtypes.float32,
+                            input_shapes=shapes,
+                            output_shapes=shapes)
+
+  def ExpectedEnginesToBuild(self, run_params):
+    """Return the expected engines to build."""
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/variables_test.py:234]: s/Return/Returns/

File: tensorflow/python/compiler/tensorrt/test/variables_test.py [hunk 23]
```diff
@@ -0,0 +1,300 @@
...
+    saved_model_dir = self._GetSavedModelDir(
+        run_params, trt_test.GraphState.ORIGINAL)
+    logging.info("Saving input SavedModel to %s", saved_model_dir)
+    save.save(my_model, saved_model_dir,
+              {signature_constants.DEFAULT_SERVING_SIGNATURE_DEF_KEY: cfunc})
+    return saved_model_dir
+
+  def ShouldRunTest(self, run_params):
+    # Variables are only supported in dynamic shape mode.
+    # TODO: dynamic engine / static engine?
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/variables_test.py:281]: s/TODO/TODO(lsugy)/

File: tensorflow/python/compiler/tensorrt/test/variables_test.py [hunk 24]
```diff
@@ -0,0 +1,300 @@
...
+
+  def GetParams(self):
+    shapes = [[8, 4, 4, 3]]
+    my_model = EngineWithoutInputsModel()
+    return self.BuildParams(my_model.__call__, dtypes.float32,
+                            input_shapes=shapes,
+                            output_shapes=shapes)
+
+  def ExpectedEnginesToBuild(self, run_params):
+    """Return the expected engines to build."""
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/variables_test.py:293]: s/Return/Returns/

**Review by bixia1 — APPROVED** [review]

File: tensorflow/python/compiler/tensorrt/test/testdata/gen_tf_variablev2_copy_model.py [hunk 25]
```diff
@@ -0,0 +1,100 @@
...
+from tensorflow.python.saved_model import builder
+from tensorflow.python.saved_model import signature_constants
+from tensorflow.python.saved_model import signature_def_utils
+from tensorflow.python.saved_model import tag_constants
+from tensorflow.python.saved_model import utils
+
+ops.disable_eager_execution()
+variable_scope.disable_resource_variables()
+
+def GenerateModelWithVariableV2_1Var2Engines(tf_saved_model_dir):
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/testdata/gen_tf_variablev2_copy_model.py:41]: There is a problem with the name of the function, 
let's s/_1/One/

P<exempt>setUp|tearDown|setUpModule|tearDownModule)|(?P<PascalCase>_?[A-Z][a-zA-Z0-9]*)|(?P<snake_case>_?[a-z][a-z0-9_]*))$' pattern [invalid-name]

File: tensorflow/python/compiler/tensorrt/test/variables_test.py [hunk 26]
```diff
@@ -0,0 +1,296 @@
...
+from tensorflow.python.framework import dtypes
+from tensorflow.python.framework import tensor_spec
+from tensorflow.python.framework import test_util
+from tensorflow.python.module import module
+from tensorflow.python.ops import array_ops
+from tensorflow.python.ops import math_ops
+from tensorflow.python.ops import nn
+from tensorflow.python.ops import variables
+from tensorflow.python.platform import tf_logging as logging
+from tensorflow.python.platform import test
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/variables_test.py:33]: this line should goes before the current line 32

File: tensorflow/python/compiler/tensorrt/test/variables_test.py [hunk 27]
```diff
@@ -0,0 +1,296 @@
...
+from tensorflow.python.saved_model import save, signature_constants
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/variables_test.py:34]: Need to use two import lines for this

Imports should be on separate lines;

File: tensorflow/python/compiler/tensorrt/trt_convert_test.py [hunk 28]
```diff
@@ -1214,5 +1219,69 @@ def testReadVariableOp(self):
...
+
+    # Check that there are two segments and that the variable is in both.
+    graph_def = graph_func.graph.as_graph_def()
+    engines = []
+    for lib_function in graph_def.library.function:
+      if re.search(r"TRTEngineOp_\d+_\d+_native_segment",
+                   lib_function.signature.name):
+        node_ops = [node.op for node in lib_function.node_def]
+        engines.append(node_ops)
+    self.assertEqual(len(engines), 2)
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/trt_convert_test.py:1276]: use assertLen instead

File: tensorflow/python/compiler/tensorrt/trt_convert_test.py [hunk 29]
```diff
@@ -1214,5 +1219,69 @@ def testReadVariableOp(self):
...
+      converter = trt_convert.TrtGraphConverterV2(
+          input_saved_model_dir=model_dir, conversion_params=conv_params,
+          use_dynamic_shape=True,
+          dynamic_shape_profile_strategy="Optimal"
+      )
+
+    # Build and save the converted model.
+    input_shapes = [[(2, 32, 32, 32, 3)]]
+
+    def input_fn():
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/trt_convert_test.py:1254]: s/input_fn/inputFn/ or similar, to use consistent naming conversion in the file

File: tensorflow/python/compiler/tensorrt/test/testdata/gen_tf_variablev2_copy_model.py [hunk 30]
```diff
@@ -0,0 +1,100 @@
...
+# limitations under the License.
+# =============================================================================
+"""Saves a TensorFlow model with a VariableV2 node used in 2 engines.
+
+   The saved model is loaded and executed by tests to check that TF-TRT can
+   successfully convert and execute the model.
+"""
+
+import numpy as np
+import tensorflow as tf
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/testdata/gen_tf_variablev2_copy_model.py:22]: import tensorflow is not allowed within tensorflow, import the module instead

**Review by bixia1 — CHANGES_REQUESTED** [review]

**Review by tfeher — COMMENTED** [review]
Thanks @Nyrio for the PR. Overall it looks good to me, I just have one question for clarification.

File: tensorflow/compiler/tf2tensorrt/convert/convert_graph.cc [hunk 31]
```diff
@@ -196,7 +197,34 @@ Status GetEngineInfo(const Graph* g,
           // Already added before.
           continue;
         }
-        VLOG(1) << "Adding const node " << input_node->name();
+        VLOG(1) << "Adding const or variable node " << input_node->name();
+      } else if (input_node->type_string() == "Identity") {
```
[the remark this answers is not in the document]
Comment (Nyrio) [reply tensorflow/compiler/tf2tensorrt/convert/convert_graph.cc:201]: If I remember correctly, the problem was that when an `Identity` node is an input of multiple nodes, some of which are not in the segment, it's not included in the segment, hence the need to copy.

**Review by bixia1 — APPROVED** [review]

File: tensorflow/python/compiler/tensorrt/test/testdata/gen_tf_variablev2_copy_model.py [hunk 32]
```diff
@@ -0,0 +1,100 @@
...
+from tensorflow.python.saved_model import builder
+from tensorflow.python.saved_model import signature_constants
+from tensorflow.python.saved_model import signature_def_utils
+from tensorflow.python.saved_model import tag_constants
+from tensorflow.python.saved_model import utils
+
+ops.disable_eager_execution()
+variable_scope.disable_resource_variables()
+
+def GenerateModelWithVariableV2_OneVarTwoEngines(tf_saved_model_dir):
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/testdata/gen_tf_variablev2_copy_model.py:41]: Need to remove _ from the name to make this a camel case

**Review by bixia1 — CHANGES_REQUESTED** [review]

**Review by bixia1 — APPROVED** [review]

**Review by bixia1 — CHANGES_REQUESTED** [review]
test fails, emailed log to author.

**Comment by Nyrio** [comment]
@bixia1 I haven't received the test logs, can you please try to send them again?

**Comment by gbaned** [comment]
Hi @Nyrio Can you please resolve conflicts? Thank you!

**Comment by Nyrio** [comment]
Thanks @bixia1 for the test logs. I have fixed the test failures and solved conflicts.

**Comment by gbaned** [comment]
Hi @Nyrio Can you please resolve conflicts? Thank you!

**Review by bixia1 — APPROVED** [review]

**Review by bixia1 — APPROVED** [review]

File: tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc [hunk 33]
```diff
@@ -659,21 +659,26 @@ void TRTEngineOp::ExecuteCalibration(OpKernelContext* ctx,
   DummyAsyncHelper dummy_async_helper;
 
   CalibrationContext* calib_ctx = cache_res->calib_ctx_.get();
-  const int num_inputs = ctx->num_inputs();
+  int num_inputs = calib_ctx->device_tensors_.size();
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc:662]: this variable is not used.

**Review by bixia1 — CHANGES_REQUESTED** [review]
emailed test failure log to author.

**Comment by Nyrio** [comment]
@bixia1 Thanks for the test logs. I added the missing `@test_util.run_v2_only` flag to the new python test and removed the unused variable.

**Review by bixia1 — APPROVED** [review]

**Review by bixia1 — CHANGES_REQUESTED** [review]
still test failure, emailed log to author.

**Comment by Nyrio** [comment]
@bixia1 Don't mind the first push. The second contains a fix for the failing tests.

**Review by bixia1 — APPROVED** [review]

**Comment by Nyrio** [comment]
@bixia1 Can you please send me the test logs?

**Comment by bixia1** [comment]
> Reference

Resent the log.

**Review by bixia1 — APPROVED** [review]

**Comment by bixia1** [comment]
Please resolve conflicts.

**Review by bixia1 — APPROVED** [review]

**Comment by Nyrio** [comment]
@bixia1 Can you please send me the logs of the internal CI build?

**Comment by poulsbo** [comment]
This is failing on Windows with "One or more test dependencies are not in the pip package."

See https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/tf_sig_build_dockerfiles/devel.usertools/code_check_full.bats

@pjannaty Does this repro for your team?

```
No message!
(in test file /usertools/code_check_full.bats, line 210)
  `exit 1' failed
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
    currently loading: tensorflow/tools/pip_package
Loading: 599 packages loaded
Loading: 601 packages loaded
Loading: 601 packages loaded
Loading: 0 packages loaded
Loading: 40 packages loaded
Loading: 40 packages loaded
Loading: 0 packages loaded
Loading: 1 packages loaded
Loading: 1 packages loaded
One or more test dependencies are not in the pip package.
If these test dependencies need to be in the TensorFlow pip package, please
add them to //tensorflow/tools/pip_package/BUILD. Otherwise, add the no_pip tag
to the test, or change code_check_full.bats in the SIG Build repository. That's
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/tf_sig_build_dockerfiles/devel.usertools/code_check_full.bats
Here are the affected tests:
For dependency //tensorflow/python/compiler/tensorrt/test:test_utils:
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
//tensorflow/python/compiler/tensorrt/test:base_test_cpu
//tensorflow/python/compiler/tensorrt/test:base_test_gpu
//tensorflow/python/compiler/tensorrt/test:batch_matmul_test_cpu
//tensorflow/python/compiler/tensorrt/test:batch_matmul_test_gpu
//tensorflow/python/compiler/tensorrt/test:biasadd_matmul_test_cpu
//tensorflow/python/compiler/tensorrt/test:biasadd_matmul_test_gpu
//tensorflow/python/compiler/tensorrt/test:binary_tensor_weight_broadcast_test_cpu
//tensorflow/python/compiler/tensorrt/test:binary_tensor_weight_broadcast_test_gpu
//tensorflow/python/compiler/tensorrt/test:bool_test_cpu
//tensorflow/python/compiler/tensorrt/test:bool_test_gpu
//tensorflow/python/compiler/tensorrt/test:cast_test_cpu
//tensorflow/python/compiler/tensorrt/test:cast_test_gpu
//tensorflow/python/compiler/tensorrt/test:combined_nms_test_cpu
//tensorflow/python/compiler/tensorrt/test:combined_nms_test_gpu
//tensorflow/python/compiler/tensorrt/test:concatenation_test_cpu
//tensorflow/python/compiler/tensorrt/test:concatenation_test_gpu
//tensorflow/python/compiler/tensorrt/test:const_broadcast_test_cpu
//tensorflow/python/compiler/tensorrt/test:const_broadcast_test_gpu
//tensorflow/python/compiler/tensorrt/test:conv2d_test_cpu
//tensorflow/python/compiler/tensorrt/test:conv2d_test_gpu
//tensorflow/python/compiler/tensorrt/test:data_dependent_shape_test_cpu
//tensorflow/python/compiler/tensorrt/test:data_dependent_shape_test_gpu
//tensorflow/python/compiler/tensorrt/test:dynamic_input_shapes_test_cpu
//tensorflow/python/compiler/tensorrt/test:dynamic_input_shapes_test_gpu
//tensorflow/python/compiler/tensorrt/test:identity_output_test_cpu
//tensorflow/python/compiler/tensorrt/test:identity_output_test_gpu
//tensorflow/python/compiler/tensorrt/test:int32_test_cpu
//tensorflow/python/compiler/tensorrt/test:int32_test_gpu
//tensorflow/python/compiler/tensorrt/test:lru_cache_test_cpu
//tensorflow/python/compiler/tensorrt/test:lru_cache_test_gpu
//tensorflow/python/compiler/tensorrt/test:memory_alignment_test_cpu
//tensorflow/python/compiler/tensorrt/test:memory_alignment_test_gpu
//tensorflow/python/compiler/tensorrt/test:multi_connection_neighbor_engine_test_cpu
//tensorflow/python/compiler/tensorrt/test:multi_connection_neighbor_engine_test_gpu
//tensorflow/python/compiler/tensorrt/test:neighboring_engine_test_cpu
//tensorflow/python/compiler/tensorrt/test:neighboring_engine_test_gpu
//tensorflow/python/compiler/tensorrt/test:quantization_test_cpu
//tensorflow/python/compiler/tensorrt/test:quantization_test_gpu
//tensorflow/python/compiler/tensorrt/test:rank_two_test_cpu
//tensorflow/python/compiler/tensorrt/test:rank_two_test_gpu
//tensorflow/python/compiler/tensorrt/test:reshape_transpose_test_cpu
//tensorflow/python/compiler/tensorrt/test:reshape_transpose_test_gpu
//tensorflow/python/compiler/tensorrt/test:test_utils
//tensorflow/python/compiler/tensorrt/test:topk_test_cpu
//tensorflow/python/compiler/tensorrt/test:topk_test_gpu
//tensorflow/python/compiler/tensorrt/test:trt_engine_op_shape_test_cpu
//tensorflow/python/compiler/tensorrt/test:trt_engine_op_shape_test_gpu
//tensorflow/python/compiler/tensorrt/test:trt_mode_test_cpu
//tensorflow/python/compiler/tensorrt/test:trt_mode_test_gpu
//tensorflow/python/compiler/tensorrt/test:unary_test_cpu
//tensorflow/python/compiler/tensorrt/test:unary_test_gpu
//tensorflow/python/compiler/tensorrt/test:variables_test_cpu
//tensorflow/python/compiler/tensorrt/test:variables_test_gpu
//tensorflow/python/compiler/tensorrt/test:vgg_block_nchw_test_cpu
//tensorflow/python/compiler/tensorrt/test:vgg_block_nchw_test_gpu
//tensorflow/python/compiler/tensorrt/test:vgg_block_test_cpu
//tensorflow/python/compiler/tensorrt/test:vgg_block_test_gpu
Loading: 11 packages loaded
Loading: 11 packages loaded
```

**Pull request closed without merging**